op-service/client/http.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	DefaultTimeoutSeconds = 30
+)
+
+type HTTP interface {
+	Get(ctx context.Context, path string, headers http.Header) (*http.Response, error)
+}
+
+type BasicHTTPClient struct {
+	endpoint string
+	log      log.Logger
+	client   *http.Client
+}
+
+func NewBasicHTTPClient(endpoint string, log log.Logger) *BasicHTTPClient {
+	// Make sure the endpoint ends in trailing slash
+	trimmedEndpoint := strings.TrimSuffix(endpoint, "/") + "/"
+	return &BasicHTTPClient{
+		endpoint: trimmedEndpoint,
+		log:      log,
+		client:   &http.Client{Timeout: DefaultTimeoutSeconds * time.Second},
+	}
+}
+
+func (cl *BasicHTTPClient) Get(ctx context.Context, p string, headers http.Header) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, cl.endpoint+p, nil)
```

Comment:
Should be prefer to use `url.Join` to handle the case when the path has a prefix `/`?

Reply:
sgtm, changed.